buildkite/bazelci.py

```diff
@@ -1331,6 +1331,13 @@ def clone_git_repository(git_repository, platform, git_commit=None):
     execute_command(["git", "submodule", "foreach", "--recursive", "git reset --hard"])
     execute_command(["git", "clean", "-fdqx"])
     execute_command(["git", "submodule", "foreach", "--recursive", "git clean -fdqx"])
+    for root, dirs, files in os.walk("."):
+        for name in files:
```

Comment:
What about using a set here?

Reply:
How would a set make things better?  This walks the tree efficiently and removes the problem files as it finds thems.

Reply:
to avoid the quadratic iteration over files:

for root, _, files in os.walk("."):
  fileset = set(files)
  if 'WORKSPACE' in fileset and '.bazelversion' in fileset:
     ...

I'm fine with either solution given that the performance differences are probably small.

Reply:
Oh duh.  It should have been  `if 'WORKSPACE' in files`.   That would make it O(2N).  Adding the set() would reduce to O(N) It's not clear what the amortized time would be, since most folders do not have WORKSPACE in them.  It would take more space.   The fastest solution would be
```
found_w = found_v = False
for file in files:
   if file == 'WORKSPACE':
     found_w = True
   if file == '.bazelversion':
     found_v = True
   if found_w and found_v:
      delete file
      break

Which is O(N) without the time to create the set hash. 

But that is moot, since we won't end up doing this.